### PR TITLE
Document provider-driven TURN credential refresh for custom providers

### PR DESCRIPTION
**Pull request by agatx**
## Problem

App-owned signaling mode (custom `SignalingProvider`, no `serverHost`) fetches ICE servers once at join. On relay-dependent networks (symmetric NAT, cross-LAN), a call that outlives its TURN credential TTL can no longer allocate a relay, so a mid-call ICE restart (e.g. after a network blip) fails with stale credentials and the call drops.

## Approach: document the existing provider-driven path (option B)

The capability to keep credentials fresh **already exists and is provider-agnostic**: a provider pushes rotated servers via `iceServersChanged` and the session applies them to all current and future peer connections — the same path the built-in `SerenadaServerProvider` uses on its `turn-refresh` loop, and already covered by per-platform contract tests. The gap was **guidance, not capability**: nothing told integrators they must drive that refresh *before* their TTL elapses.

We initially built an SDK-side refresh scheduler (option A) that fetched a TTL from the provider and refreshed at 0.8 × TTL. It worked, but the SDK can't observe a custom provider's real credential lifecycle, so it had to *guess* (rotation vs cache-hit, reported-vs-real TTL, clock domains, sleep, push-vs-poll). That guessing produced ~1000 lines and a long tail of edge cases. **The host owns its credentials, so the host should own the refresh** — exactly like the internal provider. So this PR is the documentation + a parity test, not a scheduler.

## Changes

- **`docs/sdk/sdk-customization.md`** — strengthened "ICE-server sourcing and refresh": hosts with time-limited TURN credentials must push fresh servers through `iceServersChanged` before expiry (recommend ~0.8 × TTL), the SDK applies them to all live + future peers, this mirrors the built-in `SerenadaServerProvider`, and the SDK does not refresh provider credentials for you. Spells out the failure mode and points at the existing threading guarantees.
- **`docs/sdk/sdk-api-reference.md`** — note the same on the `iceServersChanged` contract rule.
- **Web test** — asserts a provider's `iceServersChanged` push reaches the media engine (Android/iOS already have equivalent contract tests).
- **Lint cleanup** (separate commit) — eslint now honors the repo's `_`-prefix unused-binding convention + one unused import + two bare `catch` fixes; clears all 19 `no-unused-vars` errors. Remaining lint errors are pre-existing react-hooks issues in app-shell files, left for a separate change.

No SDK code change and no version bump — the contract and apply path are unchanged; this is documentation of existing behavior plus a test and a lint fix.

## Verification

Web: 412 core tests pass, `tsc` clean, 0 `no-unused-vars` lint errors. Android/iOS SDK source is unchanged (their existing `iceServersChanged` contract tests already cover the apply path).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: client/packages/core/src/SerenadaSession.ts [hunk 1]
```diff
@@ -763,6 +781,9 @@ export class SerenadaSession implements SerenadaSessionHandle {
         if (this.isInactive) {
             return;
         }
+        // Provider-driven push: apply directly without touching the refresh loop's
+        // rotation baseline, so the next poll treats these as a rotation and
+        // restarts its cycle from the fresh credentials.
         this.media.setIceServers(iceServers);
```
[the remark this answers is not in the document]
Comment (agatx) [reply client/packages/core/src/SerenadaSession.ts]: Fixed in 638c66e7. A provider-pushed `iceServersChanged` now updates the refresh loop's rotation baseline (`applyProviderIceServers`), so the later scheduled poll treats the pushed creds as the current set (cache hit) and keeps the existing, conservative expiry instead of re-anchoring from poll time. The recovery poll re-checks past that expiry and rotates onto fresh creds, so credentials are never used past expiry. This is safe now that the loop recovers (never stops) at/after expiry — the earlier reason for not touching the baseline. Applied on all 3 SDKs; web regression test added (`a provider push updates the rotation baseline so a matching poll is a cache hit`).

File: client/packages/core/src/SerenadaSession.ts [hunk 2]
```diff
@@ -820,12 +842,161 @@ export class SerenadaSession implements SerenadaSessionHandle {
...
+        this.lastAppliedIceServersSignature = this.iceServersSignature(servers);
+    }
+
+    /** Order-insensitive signature of an ICE-server list; the username/credential rotate per credential. */
+    private iceServersSignature(servers: RTCIceServer[]): string {
+        return servers
+            .map((s) => {
+                const urls = Array.isArray(s.urls) ? [...s.urls].sort().join(';') : s.urls;
+                return `${urls}|${s.username ?? ''}|${s.credential ?? ''}`;
+            })
```
[the remark this answers is not in the document]
Comment (agatx) [reply client/packages/core/src/SerenadaSession.ts]: Fixed in 638c66e7. `iceServersSignature()` now stringifies an object `credential` (a provider can return an `RTCOAuthCredential` at runtime even though the current DOM type narrows it to `string`), so OAuth-style credential rotations are no longer collapsed to `"[object Object]"` and misread as cache hits. Web regression test added (`detects rotation when only an object (OAuth) credential changes`).

**Comment by agatx** [comment]
Heads up: I force-pushed a substantially reworked branch. We pivoted from an SDK-side refresh scheduler (option A) to documenting the existing provider-driven refresh path (option B), so earlier review threads on the scheduler/TTL code no longer apply — that code is gone.

Why: the SDK can't observe a custom provider's credential lifecycle, so the scheduler had to guess (rotation vs cache-hit, reported-vs-real TTL, clocks, sleep, push-vs-poll) and grew to ~1000 lines with a long edge-case tail. The host owns its credentials, so the host owns refresh — by pushing `iceServersChanged` before its TTL, exactly like the built-in `SerenadaServerProvider`. That path already exists and is tested; the real gap was documentation. The PR is now docs + a web parity test + an unrelated lint cleanup (34 insertions / 10 deletions).
